### PR TITLE
Add Docker deployment docs

### PR DESCRIPTION
**Pull request by phaleth**
`esbuild` and `wasm-opt` optimization steps aren't explained and also there isn't any css optimization `esbuild` step included as hello world examples don't yet contain any css file, but I think this is good enough for starters. Feel free to change anything, even any of the wording used, or suggest a change.

### CODE REVIEW

File: docs/next/en-US/deploying/docker.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
+# Docker Deployment
```
Comment (arctic-hen7) [inline docs/next/en-US/deploying/docker.md:1]: Could you please apply the same changes I requested for the v0.3.x version here?

**Review by arctic-hen7 — CHANGES_REQUESTED** [review]
Really great work, thanks so much! Just some minor things on phrasing, and then I'd like all the `Dockerfile`s to be in `<details>` blocks for better scrolling support (otherwise people will have to scroll through a huge amount of code to get to what they want).

**Comment by phaleth** [comment]
I kind of just copy pasted all the suggested changes. Be aware that I can't run the website locally anymore cause of `concurrently`. The error is posted #97.

**Review by arctic-hen7 — CHANGES_REQUESTED** [review]
Just some spacing issues that maybe didn't go through properly. We need to have newlines after the `<summary>` blocks and before the closing `</details>`, otherwise the Markdown won't render properly. Otherwise great, thanks!

File: docs/next/en-US/deploying/docker.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
```
[the remark this answers is not in the document]
Comment (phaleth) [reply docs/next/en-US/deploying/docker.md:1]: Not sure what you mean here.

[the remark this answers is not in the document]
Comment (arctic-hen7) [reply docs/next/en-US/deploying/docker.md:1]: Sorry, that was meant for the whole file. All good now!

**Review by arctic-hen7 — APPROVED** [review]
Great, thanks!

**Pull request merged**